prov/efa/src/efa_cq.c

```diff
@@ -32,13 +32,16 @@ static inline uint64_t efa_cq_opcode_to_fi_flags(enum ibv_wc_opcode opcode) {
 }
 
 static void efa_cq_construct_cq_entry(struct ibv_cq_ex *ibv_cqx,
-				      struct fi_cq_tagged_entry *entry)
+				      struct fi_cq_tagged_entry *entry, int opcode)
 {
-	entry->op_context = (void *)ibv_cqx->wr_id;
-	if (ibv_cqx->wr_id)
-		entry->flags = ((struct efa_context *) ibv_cqx->wr_id)->completion_flags;
-	else
-		entry->flags = efa_cq_opcode_to_fi_flags(ibv_wc_read_opcode(ibv_cqx));
+	if (!efa_cq_wc_is_unsolicited(ibv_cqx) && ibv_cqx->wr_id) {
+		entry->op_context = (void *)ibv_cqx->wr_id;
+		entry->flags = (opcode == IBV_WC_RECV_RDMA_WITH_IMM) ? efa_cq_opcode_to_fi_flags(opcode): ((struct efa_context *) ibv_cqx->wr_id)->completion_flags;
```

Comment:
This is needed when we support `FI_RX_CQ_DATA`, and doesn't hurt for current code